### ISSUE TITLE
Argo CD loses ability to track application resources

### ISSUE THREAD

**Issue by lacarvalho91** [argoproj/argo-cd#10756]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Argo CD has seemingly lost the ability to track resources for some applications. I am using annotation based tracking but I still have the problem if I revert to label based too. On the UI I can't see any of the resources for the application, if I delete the application the resources do not get deleted and syncs get stuck waiting for hooks to complete even though they have completed - seemingly because it isn't tracking the resources. When it does get stuck, if I click refresh it seems to be able to progress to the next sync wave where it gets stuck again.

The resources for the application all seem to have the correct annotation to imply it should be being tracked.

I've tried reinstalling Argo CD, deleting all the untracked resources and recreating the app - I can't seem to get anything to work.

**To Reproduce**

I haven't been able to reproduce it locally. I am installing Argo CD via the Helm chart, when I run it locally in minikube it is fine. I only see this problem in our GKE environments.

**Expected behavior**

I expect for all applications to be able to see the resource tree on the UI (i'm only seeing the application itself with nothing else) and I expect a sync to execute sync hooks successfully and be able to complete the sync.

**Screenshots**

**Version**

```shell
argocd: v2.0.5+4c94d88
  BuildDate: 2021-07-22T20:52:31Z
  GitCommit: 4c94d886f56bcb2f9d5b3251fdc049c2d1354b88
  GitTreeState: clean
  GoVersion: go1.16
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.4.12+41f54aa
  BuildDate: 2022-09-16T00:45:01Z
  GitCommit: 41f54aa556f3ffb3fa4cf93d784fb7d30c15041c
  GitTreeState: clean
  GoVersion: go1.18.6
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v4.4.1 2021-11-11T23:36:27Z
  Helm Version: v3.8.1+g5cb9af4
  Kubectl Version: v0.23.1
  Jsonnet Version: v0.18.0

```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by agmtr** [comment]
I have the same issue when adding
kustomize.buildOptions: --enable-helm

**Comment by lacarvalho91** [comment]
When I reinstall Argo CD and recreate the application I can see some of the resources on the UI but they each have status `Unknown`

**Comment by lacarvalho91** [comment]
OK...managed to get to the bottom of this. The applications that were working properly we realised were only applications targetting external clusters, which we use GCP auth to connect to. The applications that were not working were applications syncing with the local cluster. Turned out that we had resource inclusions configured only for the local cluster and that inclusions list did not contain some of the resource types that were being sync'd. It still seems like a bit of a bug to me as from what I read on the docs that resource inclusions list should limit what resources can be discovered **and sync'd**. In my case our application was sync'ing secrets even though secrets were not configured on the inclusions list, but I guess as it couldn't then discover them the resource tree was not loading and it seemed to put the application in a weird state where it couldn't track any of its resources. I would have expected it to still be able to track the resources that were configured in the inclusions list.
